app/models/verification/residence.rb

```diff
@@ -65,6 +66,10 @@ def store_failed_attempt
     })
   end
 
+  def adult?
+    (Date.today.year - self.date_of_birth.year) > 18
```

Comment:
I don't think you need to check the JS part - just get the difference of dates *first* and then check if it's more or less than 18 years :dancer: